### PR TITLE
converge animation to moving target

### PR DESCRIPTION
**Pull request by collin**
@kand not sure how to test this case.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,3 +142,10 @@ If you'd like to see the example, clone down this code, then:
 $ cd reading-list
 $ npm run-script example
 ```
+
+## Running Tests/Development
+```bash
+npm install
+bower install
+npm test
```
Comment (kand) [inline README.md]: :+1:

File: src/jquery.reading-list.spec.js [hunk 2]
```diff
@@ -605,4 +635,10 @@ describe('Reading list', function () {
     });
   });
 
+  describe('edge cases', function () {
+    var readingList;
+
+
+  });
```
Comment (kand) [inline src/jquery.reading-list.spec.js]: empty test?

**Comment by kand** [comment]
Looks good besides the empty test :+1: 

Make sure, once you merge, to build and up the version.

File: src/jquery.reading-list.spec.js (hunk 2 above)
Comment (collin) [inline src/jquery.reading-list.spec.js]: Oof, I get the dunce cap :D, got distracted between starting the test and deciding where to write the test.

File: src/jquery.reading-list.js [hunk 3]
```diff
@@ -501,19 +501,83 @@ ReadingList.prototype.stopContainerAnimation = function () {
  * @param {Number} addPx - additional number of pixels to scroll.
  */
 ReadingList.prototype.scrollToItem = function ($item, addPx) {
+  // We need to handle image loading. On slower connections
+  // betty will load images after the reading list animated scrolling
+  // has completed. Between the end of the animation and any user-initiated
+  // motion events, it makes sense to keep the animation container
+  // locked to the top of the item.
+  //
```
Comment (collin) [inline src/jquery.reading-list.js]: @kand had to do some more things to get reading list scroll to live through loading images over the network. Tried listening directly to image loads, but there  were still some content changes slipping through the cracks.

File: bower.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "reading-list",
-  "version": "0.0.29",
+  "version": "0.0.30",
```
Comment (kand) [inline bower.json]: Please make the build and version change in master in the same commit, just so this doesn't get hidden in this PR.

File: dist/jquery.reading-list.browserify-shim.js [hunk 5]
```diff
@@ -504,19 +504,78 @@ ReadingList.prototype.stopContainerAnimation = function () {
...
+
+  var predictedScrollTop = findItemPosition();
+  var actualScrollTop = predictedScrollTop;
+  var stopContainerAnimation = this.stopContainerAnimation.bind(this);
+  var animationContainer = this.getScrollAnimationContainer();
+  var lockToItemTop = true;
 
   // ensure the animation stops when user interaction occurs
-  $document.on(MOVEMENTS, this.stopContainerAnimation.bind(this));
+  $document.on(MOVEMENTS, stopContainerAnimation);
```
Comment (kand) [inline dist/jquery.reading-list.browserify-shim.js]: I believe we want to keep this line the way it was since ```stopContainerAnimation``` should be pulled directly from the ```ReadingList``` (```this.stopContainerAnimation```) object and the element passed into the function as ```this``` should be the ```ReadingList``` object and not any other element.

File: dist/jquery.reading-list.browserify-shim.js [hunk 6]
```diff
@@ -504,19 +504,78 @@ ReadingList.prototype.stopContainerAnimation = function () {
...
+  {
+    duration: this.settings.scrollToSpeed,
+    step: function (now, tween) {
+      // images may load during the animation
+      actualScrollTop = findItemPosition();
+      tween.now = (actualScrollTop / predictedScrollTop) * tween.now;
+    },
+    complete: function () {
+      // unbind the scroll stoppage
+      $document.off(MOVEMENTS, stopContainerAnimation);
```
Comment (kand) [inline dist/jquery.reading-list.browserify-shim.js]: Same comment as above about ```stopContainerAnimation```.

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "reading-list",
-  "version": "0.0.29",
+  "version": "0.0.30",
```
Comment (kand) [inline package.json]: Same comment as the versioning comment above.

**Comment by kand** [comment]
Woops, put my comments on the ```dist``` files on accident.

File: dist/jquery.reading-list.browserify-shim.js (hunk 5 above)
Comment (collin) [inline dist/jquery.reading-list.browserify-shim.js]: The binding is maintained on this line: https://github.com/theonion/reading-list/pull/14/files#diff-4d356c5d35f141a628d40bce9248344aR549

The bound function needs to be stored as a reference so that `$document.off` can actually deregister the handler.

```javascript
> foo = function () {}
[Function]
> boundFoo = foo.bind(null)
[Function]
> boundFoo === boundFoo
true
> foo.bind(null) === foo.bind(null)
false
```

If it is being `.bind`ed in the call to `$document.on` and `$document.off`, `$document.off` will not recognize the handler, and will not remove it.

File: src/jquery.reading-list.spec.js [hunk 8]
```diff
@@ -564,6 +564,79 @@ describe('Reading list', function () {
...
+
+        $item1.position = function () {
+          return {top: 100};
+        };
+        readingList.settings.scrollToSpeed = 0;
+        readingList.scrollToItem($item1);
+        // This frame catches the initial animation
+
+        var scrollTop = sandbox.spy(readingList.$container, 'scrollTop');
+        requestAnimationFrame(function () {
```
Comment (kand) [inline src/jquery.reading-list.spec.js]: Not 100% about this one, but I don't think we want to use the actual ```requestAnimationFrame``` here since the new functionality could be tested without calling the actual function and it's time/processor dependent.

File: dist/jquery.reading-list.browserify-shim.js (hunk 5 above)
Comment (kand) [inline dist/jquery.reading-list.browserify-shim.js]: :+1:
